packages/uui-input-file/lib/uui-input-file.element.ts

```diff
@@ -0,0 +1,280 @@
+import '@umbraco-ui/uui-action-bar/lib';
+import '@umbraco-ui/uui-button/lib';
+import '@umbraco-ui/uui-icon/lib';
+import '@umbraco-ui/uui-icon-registry-essential/lib';
+import { UUIFileDropzoneElement } from '@umbraco-ui/uui-file-dropzone/lib';
+import { FormControlMixin } from '@umbraco-ui/uui-base/lib/mixins';
+
+export interface FileWrapper {
+  name: string;
+  extension: string;
+  isDirectory: boolean;
+  show: boolean;
+  size?: number;
+  thumbnail?: string;
+  source?: string;
+  file?: File;
+}
+
+/**
+ * @element uui-input-file
+ * @description - A form associated file input that supports multiple files.
+ */
+@defineElement('uui-input-file')
+export class UUIInputFileElement extends FormControlMixin(LitElement) {
+  static styles = [
+    css`
+      :host {
+        width: 100%;
+        height: 100%;
+        position: relative;
+        display: flex;
+        box-sizing: border-box;
+        border: 1px solid var(--uui-interface-border);
+      }
+
+      :host(.dropzone-active) {
+        border: 3px dashed var(--uui-color-malibu-dimmed);
+      }
+
+      #files {
+        display: grid;
+        box-sizing: border-box;
+        justify-items: center;
+        width: 100%;
+        grid-template-columns: repeat(auto-fill, minmax(150px, 1fr));
+        gap: 16px;
+        padding: 16px;
+        overflow: auto;
+        max-height: 100%;
+      }
+
+      #dropzone {
+        display: none;
+        background: var(--uui-interface-surface-alt);
+        position: absolute;
+        inset: 0px;
+        z-index: 1;
+        justify-content: center;
+        align-items: center;
+      }
+      #dropzone-content {
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        align-items: center;
+        width: 20%;
+        color: var(--uui-color-malibu-dimmed);
+      }
+
+      #dropzone-content uui-icon {
+        width: 100%;
+        height: 100%;
+      }
+      #dropzone-content span {
+        font-size: 1rem;
+        font-weight: bold;
+        white-space: nowrap;
+      }
+      #drop-button {
+        width: 100%;
+        height: 100%;
+      }
+      #add-zone {
+        width: 150px;
+        height: 150px;
+        display: flex;
+        padding: 16px;
+        box-sizing: border-box;
+        justify-content: center;
+        align-items: center;
+      }
+
+      #add-button {
+        width: 100%;
+        height: 100%;
+      }
+    `,
+  ];
+
+  @query('#dropzone')
+  private _dropZone: UUIFileDropzoneElement | undefined;
+
+  private _fileWrappers: FileWrapper[] = [];
+
+  @state()
+  get fileWrappers() {
+    return this._fileWrappers;
+  }
+  set fileWrappers(newValue) {
+    const oldValue = newValue;
+    this._fileWrappers = newValue;
+    const formData = new FormData();
+    for (const fileWrapper of this.fileWrappers) {
+      if (fileWrapper.file) {
+        formData.append(this.name, fileWrapper.file);
+      }
+    }
+    this.value = formData;
```

Comment:
You are trying to assign a `FormData` object to a value that is a union of either `string | File`. At least this can't build on my machine and the Git hook to push is failing because of it.

![image](https://user-images.githubusercontent.com/752371/160796580-b91bed41-9e79-49ee-84c0-23c5b2286243.png)